### PR TITLE
cmake: Port VVL Android CMake fixes

### PR DESCRIPTION
**Pull request by juan-lunarg**

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -14,39 +14,54 @@
...
     vktestbinding.cpp
     vktestframework.cpp
     test_environment.cpp
 )
 
-if(MSVC_IDE)
-    set_target_properties(vk_extension_layer_tests PROPERTIES VS_DEBUGGER_ENVIRONMENT "VK_LAYER_PATH=$<TARGET_FILE_DIR:VkLayer_khronos_synchronization2>")
+if (NOT ANDROID)
+    target_compile_definitions(vk_extension_layer_tests PRIVATE SHADER_OBJECT_BINARY_PATH="$<TARGET_FILE_DIR:VkLayer_khronos_shader_object>")
+    target_sources(vk_extension_layer_tests PRIVATE shader_object_tests.cpp)
```
Comment (juan-lunarg) [inline tests/CMakeLists.txt:52]: shader_object_tests.cpp currently doesn't compile for Android builds. It's currently not being tested as part of the ndk-build.

**Comment by juan-lunarg** [comment]
I'm now able to generate an APK from the CMake binaries

**Comment by juan-lunarg** [comment]
FYI @christophe-lunarg

**Comment by juan-lunarg** [comment]
APK generated by `android.py` ran successfully on my Pixel3

**Review by ziga-lunarg — APPROVED** [review]

**Review by spencer-lunarg — APPROVED** [review]

**Pull request merged**